### PR TITLE
Fix report section on original raw data (didn't contain bad channels and subject or experimenter name)

### LINKED ISSUE #930: Report section on original raw data doesn't contain bads
In my BIDS data, 3 channels are marked as bad. These show up correctly in the HTML repr for the filtered and cleaned raw, but not for the original raw. The content of the generated `*_bads.tsv` file is:

```
name	reason
P8	pre-existing (before MNE-BIDS-pipeline was run)
T7	pre-existing (before MNE-BIDS-pipeline was run)
TP10	pre-existing (before MNE-BIDS-pipeline was run)
```

Screenshot 2024-04-23 at 13 17 29

Screenshot 2024-04-23 at 13 17 34

I tried to understand what was going on but so far I don't really have a clue.

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #930

### Before merging …

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: mne_bids_pipeline/steps/preprocessing/_01_data_quality.py [hunk 1]
```diff
@@ -149,27 +149,40 @@ def assess_data_quality(
...
+                continue
+            bads_for_tsv.append(ch)
+            reasons.append(["pre-existing (before MNE-BIDS-pipeline was run)"])
 
     tsv_data = pd.DataFrame(dict(name=bads_for_tsv, reason=reasons))
     tsv_data = tsv_data.sort_values(by="name")
     tsv_data.to_csv(out_files["bads_tsv"], sep="\t", index=False)
 
     # Report
+    # Restore bads to their original state
```
Comment (hoechenberger) [inline mne_bids_pipeline/steps/preprocessing/_01_data_quality.py:183]:
```suggestion
    # Restore bads to their original state before generating the report
```

File: mne_bids_pipeline/steps/preprocessing/_01_data_quality.py [hunk 2]
```diff
@@ -149,27 +149,40 @@ def assess_data_quality(
...
+                continue
+            bads_for_tsv.append(ch)
+            reasons.append("pre-existing (before MNE-BIDS-pipeline was run)")
 
     tsv_data = pd.DataFrame(dict(name=bads_for_tsv, reason=reasons))
     tsv_data = tsv_data.sort_values(by="name")
     tsv_data.to_csv(out_files["bads_tsv"], sep="\t", index=False)
 
     # Report
+    # Restore bads to their original state before generating the report
```
Comment (hoechenberger) [inline mne_bids_pipeline/steps/preprocessing/_01_data_quality.py:183]:
```suggestion
    # Restore bads to their original state so they will show up in the report
```

File: mne_bids_pipeline/_report.py [hunk 3]
```diff
@@ -848,9 +848,9 @@ def _add_raw(
     cfg: SimpleNamespace,
     report: mne.report.Report,
     bids_path_in: BIDSPath,
+    raw: BaseRaw,
```
Comment (hoechenberger) [inline mne_bids_pipeline/_report.py:851]: This change should avoid the problem we saw here, as it now forces us to provide pre-loaded raw data (with hopefully the correct metadata set). previously this was optional. see also the changes below

**Comment by hoechenberger** [comment]
Circle fails for ds000246 due to outdated cache

**Comment by hoechenberger** [comment]
> I'll fix the caches in #929

Thank you!

**Comment by larsoner** [comment]
> Should we only set the pre-existing bads when generating the HTML repr for the original data, or should we set the pre-existing bads + all bads from the data quality checks?

To me the optimal thing to do here I think is to have two report entries:

1. One which is the data as it comes originally from the dataset author with bads from the sidecar. It tells you what you start with before doing anything to it with the pipeline
2. Another which is the data after our autobads (this section only needs to exist if we do some autobad detection). It tells you what the automated approaches did to the data

To me it's okay for brevity to have (1) have all the nice raw data plots, and have (2) maybe have less -- like just a list of channels removed or something? I think we're already kind of close to this organization:

image

But from that data quality plot it's not immediately clear which channels were added as bad, if any. Even just a HTML entry that lists the bads (when bad enabled) and flats (when flat enabled) would be a nice help

**Comment by larsoner** [comment]
... and organizationally I think it might make sense to call that section in the sidebar `Raw (autobad)` or similar. It makes it clear the parentheticals from original->autobad->maxwell->filtered->cleaned. "Data Quality" is an outlier at the moment in that it uses the step name

**Comment by hoechenberger** [comment]
Thanks for sharing your ideas, they resonate well with me!

I believe then at least the bits of code that this PR touches should be good! And the rest could be addressed in future PRs

**Comment by larsoner** [comment]
I merged `main` into the branch, you should be able to add to the changelog now and it should come back green

**Comment by hoechenberger** [comment]
Thanks @larsoner, I pushed changelog entries and enabled auto-merge

**Pull request merged**